pyradiosky/skymodel.py

```diff
@@ -629,6 +630,81 @@ def __eq__(self, other, check_extra=True):
 
         return equal
 
+    def evaluate_stokes(
+        self, freqs, inplace=True, freq_interp_kind="cubic", run_check=True,
+    ):
+        """
+        Evaluate the stokes array to specified frequencies.
+
+        Produces a SkyModel object that is in the `full` frequency spectral type, based on
+        the current spectral type:
+        - full: Extract a subset of existing frequencies.
+        - subband: Interpolate to new frequencies.
+        - spectral_index: Evaluate at the new frequencies.
+        - flat: Copy to new frequencies.
+
+        Parameters
+        ----------
+        freqs: Quantity
+            Frequencies at which Stokes parameters will be evaluated.
+        inplace: bool
+            If True, modify the current SkyModel object.
+            Otherwise, returns a new instance. Default True.
+        freq_interp_kind: str or int
+            Spline interpolation order, as can be understood by scipy.interpolate.interp1d.
+            Defaults to 'cubic'
+        run_check: bool
+            Run check on new SkyModel.
+            Default True.
+        """
+        freqs = np.atleast_1d(freqs)
+
+        if inplace:
+            sky = self
+        else:
+            sky = copy.deepcopy(self)
```

Comment:
Actually, SkyModel has it's own copy method (because UVBase has one) that does a deepcopy. So I'd replace this line with `sky=self.copy()`

Reply:
Good to know! I've replaced the line as suggested, and in the test.